selfdrive/boardd/pandad.py

```diff
@@ -74,10 +74,10 @@ def panda_sort_cmp(a: Panda, b: Panda):
 
   # sort by hardware type
   if a_type != b_type:
-    return a_type < b_type
+    return -1 if a_type < b_type else 1
```

Comment:
perhaps we should sort with `(p.is_internal(), p.get_type(), p.get_usb_serial())` instead of this comparison function

Reply:
It will be simpler! I changed the implementation as you suggested. It turned out that the first key should be the `not p.is_internal()` because internal panda should be first.